### PR TITLE
[WIP] support multi Istio deployments in same k8s cluster 

### LINKED ISSUE #31115: Multi-Istio deployments in same k8s cluster
**Describe the feature request**

After issue #26679 resolved, istiod can be configured to watch a subset of namespaces in a k8s cluster by using `meshConfig.discoverySelectors`. But for deploying multi-istio in a k8s cluster, there are still some gaps:

- namespace controller injects ConfigMap istio-root-ca-cert to namespaces, need to scoping it to the namespaces which are part of mesh managed by one Istio deployment, otherwise, it would override istio-root-ca-cert injected by another Istiod deployment.
- mutating/validating webhook should be configured to run on an object based on which Istio deployment manages its namespace.
- secrets controller watches all namespaces currently, it should also be configured to avoid trigger XDS push unnecessarily when a Secret doesn't belong to a Istio deployment. This is not mandatory, but better to have.

PR #29802 implemented a DiscoveryNamespacesFilter, which can be reused by namespace controller and secrets controller, but looks like a different set of namespace selectors should be defined, because "discovery - namespaces watched by Istio" and "namespaces in a mesh" are different concept.

Per my understanding, another namespace selector("meshNamespaceSelectors"?) in MeshConfig should be defined and items listed above should be modified accordingly. Please correct me if I misunderstand anything.

**Describe alternatives you've considered**

[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

[] Multi Cluster
[ ] Virtual Machine
[X] Multi Control Plane

**Additional context**

### PR DESCRIPTION
**Pull request by liuxu623**
**Please provide a description of this PR:**
Only create istio-ca-root-cert configmap in namesapces which select by discoverySelectors.
Fix #31115

### CODE REVIEW

**Review by bianpengyuan — COMMENTED** [review]
discoverySelector is meant for service discovery filtering, not mesh enrollment. I think it is a perfectly valid use case that some namespaces are enrolled into the mesh (i.e. have sidecar injected), but not included in the discoverySelector. I think we will need an explicit option to control root cert configmap selector instead of piggyback on discoverySelector.

**Comment by liuxu623** [comment]
https://github.com/istio/api/pull/2212 add namespaceSelectors

**Comment by sha-rath** [comment]
Not stale
cc @ericvn

**Comment by kfaseela** [comment]
@liuxu623  are you planning to proceed further with this PR? Else, I am interested in picking this up. Please let me know

File: pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go [hunk 1]
```diff
@@ -62,6 +68,8 @@ func NewNamespaceController(kubeClient kube.Client, caBundleWatcher *keycertbund
 	c.namespacesInformer = kubeClient.KubeInformer().Core().V1().Namespaces().Informer()
 	c.namespaceLister = kubeClient.KubeInformer().Core().V1().Namespaces().Lister()
 
+	c.namespaceFilter = filter.NewDiscoveryNamespacesFilter(c.namespaceLister, options.MeshWatcher.Mesh().NamespaceSelectors)
```
[the remark this answers is not in the document]
Comment (liuxu623) [reply pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go:72]: Yes, you are right, we need support mesh config change.

**Pull request closed without merging**